core/src/transaction_scheduler.rs

```diff
@@ -0,0 +1,1913 @@
+                    return Err(SchedulerError::AccountBlocked(**acc));
+                }
+            }
+
+            // reads are blocked if the account exists in write blocked state
+            for acc in readonly_keys {
+                if highest_wl_blocked_account_fees.get(*acc).is_some() {
+                    trace!("read locked account blocked by another tx: {:?}", acc);
+                    return Err(SchedulerError::AccountBlocked(**acc));
+                }
+            }
+        }
+
+        // double check to make sure we can lock this against currently executed transactions and
+        // accounts
+        Self::can_lock_accounts(account_locks, writable_keys, readonly_keys)?;
+
+        Ok(())
+    }
+
+    fn get_scheduled_batch(
+        unprocessed_packets: &mut UnprocessedPacketBatches,
+        scheduled_accounts: &Arc<Mutex<AccountLocks>>,
+        num_txs: usize,
+        bank: &Arc<Bank>,
+        qos_service: &QosService,
+        config: &TransactionSchedulerConfig,
+    ) -> (Vec<SanitizedTransaction>, Vec<DeserializedPacket>) {
+        let mut sanitized_transactions = Vec::with_capacity(num_txs);
+        let mut rescheduled_packets = Vec::with_capacity(1_000);
+
+        // hashmap representing the highest fee of currently write-locked and read-locked blocked accounts
+        // almost a pseudo AccountLocks but fees instead of hashset/read lock count
+        let mut highest_wl_blocked_account_fees = HashMap::with_capacity(10_000);
+        let mut highest_rl_blocked_account_fees = HashMap::with_capacity(10_000);
+
+        let mut lookback_count = 0;
+
+        while let Some(deserialized_packet) = unprocessed_packets.pop_max() {
+            match Self::try_schedule(
+                &deserialized_packet,
+                bank,
+                &mut highest_wl_blocked_account_fees,
+                &mut highest_rl_blocked_account_fees,
+                scheduled_accounts,
+                qos_service,
+                config,
+            ) {
+                Ok(sanitized_tx) => {
+                    sanitized_transactions.push(sanitized_tx);
+                    if sanitized_transactions.len() >= num_txs {
+                        break;
+                    }
+                }
+                Err(e) => {
+                    trace!("e: {:?}", e);
+                    match e {
+                        SchedulerError::InvalidSanitizedTransaction
+                        | SchedulerError::InvalidTransactionFormat(_)
+                        | SchedulerError::TransactionCheckFailed(_) => {
+                            // non-recoverable error, drop the packet
+                            continue;
+                        }
+                        SchedulerError::AccountInUse | SchedulerError::AccountBlocked(_) => {
+                            // need to reschedule
+                            // error!("e: {}", e);
+                            rescheduled_packets.push(deserialized_packet);
```

Comment:
need to optimize this pop then push a little more. something about keeping blocked accounts somewhere else + linking rw accounts easily w/ fees

Reply:
actually ill try what banking stage does with the memswap + remove then swap back.

might be worth looking at skiplist too, but can look at that later. https://docs.rs/skiplist/latest/skiplist/ordered_skiplist/struct.OrderedSkipList.html